beacon_node/http_api/src/lib.rs

```diff
@@ -1022,6 +1023,116 @@ pub fn serve<T: BeaconChainTypes>(
+                        // will be deprecated from the beacon API. This will entail creating full blocks in
+                        // `validator/blinded_blocks`, caching their payloads, and transforming them into blinded
+                        // blocks. We will access the payload of those blocks here. This flow should happen if the
+                        // execution layer has no payload builders or if we have not yet finalized post-merge transition.
+                        let payload = el
+                            .block_on(|el| el.propose_blinded_beacon_block(&block))
+                            .map_err(|e| {
+                                warp_utils::reject::custom_server_error(format!(
+                                    "proposal failed: {:?}",
+                                    e
+                                ))
+                            })?;
+                        let new_block = SignedBeaconBlock::Merge(SignedBeaconBlockMerge {
+                            message: BeaconBlockMerge {
+                                slot: block.message().slot(),
+                                proposer_index: block.message().proposer_index(),
+                                parent_root: block.message().parent_root(),
+                                state_root: block.message().state_root(),
+                                body: BeaconBlockBodyMerge {
+                                    randao_reveal: block.message().body().randao_reveal().clone(),
+                                    eth1_data: block.message().body().eth1_data().clone(),
+                                    graffiti: *block.message().body().graffiti(),
+                                    proposer_slashings: block
+                                        .message()
+                                        .body()
+                                        .proposer_slashings()
+                                        .clone(),
+                                    attester_slashings: block
+                                        .message()
+                                        .body()
+                                        .attester_slashings()
+                                        .clone(),
+                                    attestations: block.message().body().attestations().clone(),
+                                    deposits: block.message().body().deposits().clone(),
+                                    voluntary_exits: block
+                                        .message()
+                                        .body()
+                                        .voluntary_exits()
+                                        .clone(),
+                                    sync_aggregate: block
+                                        .message()
+                                        .body()
+                                        .sync_aggregate()
+                                        .unwrap()
```

Comment:
This `unwrap` isn't ideal, but I note that you've fixed it in #3134 and it isn't reachable from non-blinded block proposal routines, so I think we're good :+1:

Reply:
Whoops, oversight on my part!